tasks/post-bigquery-metrics-task.yaml

```diff
@@ -0,0 +1,122 @@
+    image: gcr.io/google.com/cloudsdktool/google-cloud-cli:stable@sha256:767516bf8e50f22c074c07e1138844744b80345e5ca315ea413a88ab22e36089
+    volumeMounts:
+    - name: gcp-service-account
+      mountPath: /mnt/gcp-service-account
+    env:
+    - name: PIPELINE_RUN_NAME
+      # E.g. "scanner-v4-db-on-push-477w8"
+      valueFrom:
+        fieldRef:
+          fieldPath: metadata.labels['tekton.dev/pipelineRun']
+    - name: ORIGINAL_PIPELINE_RUN_NAME
+      # E.g. "scanner-v4-db-on-push"
+      valueFrom:
+        fieldRef:
+          fieldPath: metadata.labels['pipelinesascode.tekton.dev/original-prname']
+    - name: REPO_URL
+      # E.g. "https://github.com/stackrox/stackrox"
+      valueFrom:
+        fieldRef:
+          fieldPath: metadata.annotations['pipelinesascode.tekton.dev/repo-url']
+    - name: SOURCE_BRANCH
+      # E.g. "konflux/references/master"
+      valueFrom:
+        fieldRef:
+          fieldPath: metadata.annotations['pipelinesascode.tekton.dev/source-branch']
+    - name: PULL_REQUEST_NUMBER
+      # E.g. "14577"
+      valueFrom:
+        fieldRef:
+          fieldPath: metadata.labels['pipelinesascode.tekton.dev/pull-request']
+    - name: COMMIT_SHA
+      # E.g. "2bb555465e2b822abb4eb13da2167908670ff5c9"
+      valueFrom:
+        fieldRef:
+          fieldPath: metadata.labels['pipelinesascode.tekton.dev/sha']
+    script: |
+      #!/usr/bin/env bash
+
+      set -euo pipefail
+
+      GCP_PROJECT="acs-san-stackroxci"
+      TABLE_NAME="${GCP_PROJECT}.ci_metrics.stackrox_jobs"
+
+      gcloud auth activate-service-account --key-file /mnt/gcp-service-account/konflux-metrics-gcp-service-account
+      gcloud config set project "${GCP_PROJECT}"
+
+      AGGREGATE_TASKS_STATUS="$(params.AGGREGATE_TASKS_STATUS)"
+
+      bq query \
+        --use_legacy_sql=false \
+        --parameter="id::${PIPELINE_RUN_NAME}" \
+        --parameter="name::${ORIGINAL_PIPELINE_RUN_NAME}" \
+        --parameter="repo::${REPO_URL}" \
+        --parameter="branch::${SOURCE_BRANCH}" \
+        --parameter="pr_number:INTEGER:${PULL_REQUEST_NUMBER:-NULL}" \
+        --parameter="commit_sha::${COMMIT_SHA}" \
+        --parameter="outcome::${AGGREGATE_TASKS_STATUS:-NULL}" \
+        "
+        -- Create a temporary table for holding a record for this job with the same schema as the target table.
+        CREATE TEMP TABLE _SESSION.this_job AS SELECT * FROM ${TABLE_NAME} LIMIT 0;
+
+        -- Add the job's record into the temporary table.
+        INSERT INTO _SESSION.this_job
+          (id, name, repo, branch, pr_number, commit_sha, started_at, outcome, ci_system)
+          VALUES
+          (@id, @name, @repo, @branch, @pr_number, @commit_sha, CURRENT_TIMESTAMP(), @outcome, 'konflux');
+
+        -- If the outcome is provided, set the job's end timestamp to "now" which is recorded in started_at.
+        UPDATE _SESSION.this_job
+        SET stopped_at = CASE WHEN outcome IS NULL THEN NULL ELSE started_at END
```

Comment:
[nit] Can we work with formatting to make this easier to read or will this break the ternary operator?

Reply:
Added newlines. Please suggest your version if the new one doesn't look good yet.